.travis/deploy-acs.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+
+set -e
+# Utility Functions
+
+# pod status
+pod_status() {
+  kubectl get pods --namespace $namespace -o=custom-columns=NAME:.metadata.name,STATUS:.status.phase,READY:.status.conditions[?\(@.type==\'Ready\'\)].status
+}
+
+# pods ready
+pods_ready() {
+  PODS_COUNTER=0
+  PODS_COUNTER_MAX=60
+  PODS_SLEEP_SECONDS=10
+
+  while [ "$PODS_COUNTER" -lt "$PODS_COUNTER_MAX" ]; do
+    totalpods=$(pod_status | grep -v NAME | wc -l | sed 's/ *//')
+    readypodcount=$(pod_status | grep ' True' | wc -l | sed 's/ *//')
+    if [ "$readypodcount" -eq "$totalpods" ]; then
+          echo "     $readypodcount/$totalpods pods ready now"
+          pod_status
+      echo "All pods are ready!"
+      break
+    fi
+      PODS_COUNTER=$((PODS_COUNTER + 1))
+      echo "just $readypodcount/$totalpods pods ready now - sleeping $PODS_SLEEP_SECONDS seconds - counter $PODS_COUNTER"
+      sleep "$PODS_SLEEP_SECONDS"
+      continue
+    done
+  
+  if [ "$PODS_COUNTER" -ge "$PODS_COUNTER_MAX" ]; then
+    pod_status
+    echo "Pods did not start - exit 1"
+    exit 1
+  fi
+}
+
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: $namespace
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: $namespace:psp
+  namespace: $namespace
+rules:
+- apiGroups:
+  - policy
+  resourceNames:
+  - kube-system
+  resources:
+  - podsecuritypolicies
+  verbs:
+  - use
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: $namespace:psp:default
+  namespace: $namespace
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: $namespace:psp
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: $namespace
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: $namespace:psp:$release_name_ingress-nginx-ingress
+  namespace: $namespace
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: $namespace:psp
+subjects:
+- kind: ServiceAccount
+  name: $release_name_ingress-nginx-ingress
+  namespace: $namespace
+---
+$(kubectl create secret docker-registry quay-registry-secret --dry-run=client --docker-server=${DOCKER_REGISTRY} --docker-username=${DOCKER_REGISTRY_USERNAME} --docker-password=${DOCKER_REGISTRY_PASSWORD} -n $namespace -o yaml)
+EOF
+
+# install ingress
+helm upgrade --install $release_name_ingress stable/nginx-ingress \
```

Comment:
Is there a reason for using "upgrade --install" rather than just "helm install"? Is it to support re-started builds?

Reply:
exactly, to support restarted jobs, also the first components are created with kubectl apply for the same reasons.